### PR TITLE
Add automatic page date fallback

### PR DESCRIPTION
**Pull request by vbatoufflet**
Hi,

This PR adds an automatic value for `Page.Date` using the file's `ModTime` as fallback when none is provided?

Please let me know if it suits to you.

Regards.

### CODE REVIEW

**Comment by bep** [comment]
I guess this is a fair compromise.

File: hugolib/page.go [hunk 1]
```diff
@@ -663,6 +664,13 @@ func (p *Page) update(f interface{}) error {
 		p.Draft = !*published
 	}
 
+	if p.Date.IsZero() {
+		fi, err := os.Stat(filepath.Join(helpers.AbsPathify(viper.GetString("ContentDir")), p.File.Path()))
```
Comment (bep) [inline hugolib/page.go]: You need to use the https://github.com/spf13/hugo/blob/master/hugofs/fs.go#L30 filesystem.

**Comment by vbatoufflet** [comment]
Hi,

Thanks for the feedback. I just updated my PR to use hugofs.

**Comment by luixxiul** [comment]
This causes a trouble if you are using Page in terms of WordPress: about/sitemap/privacy policy/etc which do not have ``.Date`` suddenly start to appear in feeds or the list of recent posts.

**Comment by vbatoufflet** [comment]
Hi all,

Maybe we can define a configuration option to toggle automatic date feature?

WDYT?

**Comment by bep** [comment]
>WDYT?

Yes, we need that, and set it to default = false.

**Comment by luixxiul** [comment]
That would be great!

2016/07/01 16:25 "Bjørn Erik Pedersen" <notifications@github.com>:

**Comment by vbatoufflet** [comment]
Yes, I will create a PR.

**Comment by vbatoufflet** [comment]
Hi All,

Any suggestion on the setting name, e.g. `AutomaticFallbackDates` or `AutoDates`?

**Comment by bep** [comment]
@vbatoufflet see my comment on https://github.com/spf13/hugo/issues/2244

**Pull request closed without merging**